index.bs

```diff
@@ -169,11 +207,71 @@ This allows developers to detect support for paint timing for a particular [=bro
+This request can be the <a>associated image request</a> of an {{Element}}.
+
+Note: The current language is vague since it does not point to specific algorithms.
+This can be made more rigorous when the corresponding processing models have a more unified processing model.
+
+When an {{Element}} <em>element</em>'s <a>associated image request</a> has become <a>completely available</a>, run the algorithm to <a>process an image that finished loading</a> passing in <em>element</em> and its <a>associated image request</a> as inputs.
+
+<h4 id="sec-modifications-CSS">Modifications to the CSS specification</h4>
+
+When the user agent is executing the <a>painting order</a>, it must populate the <a>set of owned text nodes</a> of the painted {{Element|Elements}} so that the following is true:
+
+<div algorithm="text aggregation">
+    * If a {{Text}} object |text| will not be painted due to the font face being in its <a>font block period</a>, then it is not <a for="set">appended</a> to the <a>set of owned text nodes</a> of any {{Element}}.
+    * Otherwise, |text| is <a for="set">appended</a> to the <a>set of owned text nodes</a> of the {{Element}} which determines the <a>containing block</a> of |text|.
+</div>
+
+NOTE: A user agent might want to use a stack to efficiently compute the <a>set of owned text nodes</a> while implementing the <a>painting order</a>.
+
+Every {{Element}} has a list of <dfn>associated background image requests</dfn> which is initially an empty array.
+When the processing model for the {{Element}} <em>element</em>'s style requires a new image resource (to be displayed as background image), the <a>image request</a> created by the new resource is appended to <em>element</em>'s <a>associated background image requests</a>.
+Whenever an <a>image request</a> in an {{Element}} <em>element</em>'s <a>associated background image requests</a> becomes <a>completely available</a>, run the algorithm to <a>process an image that finished loading</a> with <em>element</em> and <a>image request</a> as inputs.
+
+NOTE: we assume that there is one <a>image request</a> for each {{Element}} that a <a>background-image</a> property affects and for each URL that the <a>background-image</a> property specifies.
+So, for example, if there is a style with two URLs affecting all <{div}>s, and there are two <{div}>s, then there will be four <a>image requests</a>.
+This means that a single <a>background-image</a> property could produce multiple timings, because it can affect multiple elements and because it can specify multiple URLs.
+
+<h4 dfn>Process image that finished loading</h4>
+
+<div algorithm="image element loaded">
+To <dfn>process an image that finished loading</dfn> given |element| and |imageRequest| as inputs:
+    1. Let |element| be the input {{Element}}.
+    1. Let |imageRequest| be |element|'s <a>associated image request</a>.
```

Comment:
Isn't it an input?

Reply:
Yup. And it shouldn't even be the associated image request; it might be the background instead.
For that matter, |element| is an input to this, too.